components/user-store/org.wso2.carbon.identity.user.store.count/src/main/java/org/wso2/carbon/identity/user/store/count/UserStoreCountService.java

```diff
@@ -181,27 +163,87 @@ public Long countRolesInDomain(String filter, String domain) throws UserStoreCou
      * @param valueFilter filter for the claim values
      * @return the number of users matching the given claim and filter within this user store domain
      */
-    public Long countByClaimInDomain(String claimURI, String valueFilter, String domain) throws
-            UserStoreCounterException {
-
-        UserStoreCountRetriever counter = UserStoreCountUtils.getCounterInstanceForDomain(domain);
-        if (counter != null) {
-            return counter.countClaim(claimURI, valueFilter);
-        } else {
-            return Long.valueOf(-1);
-        }
+    public long countByClaimInDomain(String claimURI, String valueFilter, String domain)
+            throws UserStoreCounterException {
 
+        String filterWithDomain = getFilterWithDomain(domain, valueFilter);
+        return getUserCountWithClaims(claimURI, filterWithDomain);
     }
 
     /**
      * Get count enabled user stores.
+     *
      * @return
      * @throws UserStoreCounterException
      */
     public String[] getCountEnabledUserStores() throws UserStoreCounterException {
+
         Set<String> domains = UserStoreCountUtils.getCountEnabledUserStores();
         return domains.toArray(new String[domains.size()]);
 
     }
 
+    /**
+     * Get User count.
+     *
+     * @param claimURI    claim uri
+     * @param valueFilter filter
+     * @return user count
```

Comment:
Missing return description, and Full stops at the end. Shall we add more details for the valueFilter parameter?